### PR TITLE
[NEW] MAJ tcpdi parse library

### PR DESCRIPTION
**Pull request by kkhelifa-opendsi**
[NEW] Update tcpdi parse library

### CODE REVIEW

File: htdocs/includes/tcpdi/tcpdi_parser.php [hunk 1]
```diff
@@ -484,7 +484,7 @@ protected function decodeXrefStream($startxref, $xref=array()) {
 			$v = $sarr[$key];
 			if (($key == '/Type') AND ($v[0] == PDF_TYPE_TOKEN AND ($v[1] == 'XRef'))) {
 				$valid_crs = true;
-			} elseif (($key == '/Index') AND ($v[0] == PDF_TYPE_ARRAY AND count($v[1]) >= 2)) {
+            } elseif (($key == '/Index') AND ($v[0] == PDF_TYPE_ARRAY AND count($v[1] >= 2))) {
```
Comment (eldy) [inline htdocs/includes/tcpdi/tcpdi_parser.php:487]: This looks strange
AND count($v[1] >= 2))) {
Old code
AND count($v[1]) >= 2)) {

Is it a fix on your own or official project patch ?

**Comment by eldy** [comment]
Is it a fix on your own or official project patch ? If project patch, where does he come from ?

File: htdocs/includes/tcpdi/tcpdi_parser.php (hunk 1 above)
Comment (kkhelifa-opendsi) [reply htdocs/includes/tcpdi/tcpdi_parser.php:487]: it's come from here https://github.com/pauln/tcpdi_parser

Comment (eldy) [reply htdocs/includes/tcpdi/tcpdi_parser.php:487]: The old code was a fix when making lib compatible with php8. So i just think this fix was not included into the version at pauln. But i think we must keep it.
Can you try to also add this fix 
AND count($v[1]) >= 2)) {
And tell if your use of tcpdfi still works for the case you tested ?

**Comment by eldy** [comment]
@kkhelifa-opendsi 
Can you try to also add this fix
`} elseif (($key == '/Index') AND ($v[0] == PDF_TYPE_ARRAY AND count($v[1]) >= 2)) {`   (old code)
instead of 
`} elseif (($key == '/Index') AND ($v[0] == PDF_TYPE_ARRAY AND count($v[1] >= 2))) {`   (new one with change on parenthesis for the count)
And tell if your use of tcpdfi still works for the case you tested ?

**Comment by eldy** [comment]
I think tcpdf was updated since , so i close the pr.

**Pull request closed without merging**